src/shared/features/AppletSettings/ExportDataSetting/ExportDataSetting.test.tsx

```diff
@@ -58,4 +72,72 @@ describe('ExportDataSetting', () => {
 
     expect(mockOnClose).toHaveBeenCalled();
   });
+
+  describe('should pass settings specified in settings popup to the export popup', () => {
+    test.each`
+      exportType                  | expectedFromTime            | description
+      ${ExportDateType.AllTime}   | ${new Date(createdDate)}    | ${'use applet create time and now for all time'}
+      ${ExportDateType.Last24h}   | ${addDays(new Date(), -1)}  | ${'use correct dates for last 24h'}
+      ${ExportDateType.LastWeek}  | ${addDays(new Date(), -7)}  | ${'use correct dates for last week'}
+      ${ExportDateType.LastMonth} | ${addDays(new Date(), -30)} | ${'use correct dates for last month'}
+    `('$description', async ({ exportType, expectedFromTime }) => {
+      const mockOnClose = jest.fn();
+
+      jest.spyOn(encryptionFunctions, 'getAppletEncryptionInfo').mockImplementation(() =>
+        Promise.resolve({
+          getPublicKey: getPublicKeyMock,
+        }),
+      );
+
+      renderWithProviders(
+        <ExportDataSetting isExportSettingsOpen onExportSettingsClose={mockOnClose} />,
+        {
+          preloadedState,
+        },
+      );
+
+      await waitFor(() =>
+        expect(screen.queryByTestId(DATA_TESTID_EXPORT_DATA_SETTINGS_POPUP)).toBeInTheDocument(),
+      );
+
+      const dateType = screen.getByTestId(`${DATA_TESTID_EXPORT_DATA_SETTINGS_POPUP}-dateType`);
+      expect(dateType).toBeVisible();
+      const input = dateType.querySelector('input');
+      input && fireEvent.change(input, { target: { value: exportType } });
```

Comment:
Please use `userEvent` library to simulate an user interaction more closer, if it applicable. For ex., `await userEvent.type` instead of  `fireEvent.change`,   `await userEvent.click`  instead of `fireEvent.click` here and below.

Reply:
Sounds good, I updated all of the `.click` instances to `userEvent.click`. I had trouble updating the `fireEvent.change` instances because the input is actually a hidden input in the Material UI element that has `pointer-events: none`, so `.type` could not be used. I tried triggering the Material UI dropdown as well with a variety of `userEvent.click` but was having trouble locating it/getting the dropdown to work correctly.

If it's okay for now, I'd like to leave it to use `fireEvent.change`, though suggestions are very welcome

Reply:
I found a flag to change the `pointereventscheck` logic https://testing-library.com/docs/user-event/options/#pointereventscheck. Personally, I didn't check it, but it guess, it should work.